pkg/templates/antiaffinity/template.go

```diff
@@ -75,20 +75,22 @@ func init() {
 				for _, preferred := range preferredAffinity {
 					err := validateAffinityTermMatchesAgainstNodes(preferred.PodAffinityTerm,
 						podTemplateSpec.Namespace, podTemplateSpec.Labels, topologyKeyMatcher)
-					if err != nil {
-						foundIssues = append(foundIssues, diagnostic.Diagnostic{
-							Message: err.Error(),
-						})
+					if err == nil {
+						return nil
 					}
+					foundIssues = append(foundIssues, diagnostic.Diagnostic{
+						Message: err.Error(),
+					})
 				}
 				for _, required := range requiredAffinity {
 					err := validateAffinityTermMatchesAgainstNodes(required, podTemplateSpec.Namespace,
 						podTemplateSpec.Labels, topologyKeyMatcher)
-					if err != nil {
-						foundIssues = append(foundIssues, diagnostic.Diagnostic{
-							Message: err.Error(),
-						})
+					if err == nil {
+						return nil
 					}
+					foundIssues = append(foundIssues, diagnostic.Diagnostic{
+						Message: err.Error(),
+					})
 				}
 				return foundIssues
```

Comment:
This will return `nil` if no issues are found, which is wrong. (Which also suggests we need some unit tests here 😅)

Reply:
Hm, but what would the expected behavior be if we didn't find any issues?

We are already short-circuiting if no anti-affinity rule is created beforehand.

Reply:
Ah, just saw that short-circuit. Well, we need to enhance it so that it also handles the case where podAntiAffinity is not `nil`, but both the arrays ("required" and "preferred") are empty. If we short-circuit in that case as well, then we should be good.